hr_expense_operating_unit/tests/test_hr_expense_operating_unit.py

```diff
@@ -0,0 +1,113 @@
+
+        # Main Operating Unit
+        self.ou1 = self.env.ref('operating_unit.main_operating_unit')
+        # B2C Operating Unit
+        self.b2c = self.env.ref('operating_unit.b2c_operating_unit')
+
+        self.user1 = self._create_user('Test HR User 1', 'user_1', 'demo1',
+                                       [self.grp_hr_user, self.grp_accou_mng,
+                                        self.grp_account_invoice],
+                                       self.company,
+                                       [self.ou1, self.b2c])
+        self.user2 = self._create_user('Test HR User 2', 'user_2', 'demo2',
+                                       [self.grp_hr_user, self.grp_accou_mng,
+                                        self.grp_account_invoice],
+                                       self.company,
+                                       [self.b2c])
+
+        self.emp = self._create_hr_employee()
+
+        self.hr_expense1 = self._create_hr_expense(
+            self.ou1, self.emp)
+
+        self.hr_expense2 = self._create_hr_expense(
+            self.b2c, self.emp)
+
+        self._post_journal_entries(self.hr_expense1)
+        self._post_journal_entries(self.hr_expense2)
+
+    def _create_user(self, name, login, pwd, groups, company, operating_units,
+                     context=None):
+        """Creates a user."""
+        group_ids = [group.id for group in groups]
+        user = self.res_users_model.create({
+            'name': name,
+            'login': login,
+            'password': pwd,
+            'email': 'example@yourcompany.com',
+            'company_id': company.id,
+            'company_ids': [(4, company.id)],
+            'operating_unit_ids': [(4, ou.id) for ou in operating_units],
+            'groups_id': [(6, 0, group_ids)]
+        })
+        return user
+
+    def _create_hr_employee(self):
+        """Creates an Employee."""
+        emp = self.hr_employee_model.create({
+            'name': "Test Employee",
+            'address_home_id': self.partner1.id,
+            })
+        return emp
+
+    def _create_hr_expense(self, operating_unit, emp):
+        """Creates Expense for employee."""
+        expense = self.hr_expense_model.create({
+            'name': " Food Expense ",
+            'product_id': self.product1.id,
+            'operating_unit_id': operating_unit.id,
+            'unit_amount': '10.0',
+            'quantity': '5',
+            'employee_id': emp.id
+            })
+        return expense
+
+    def _post_journal_entries(self, expense):
+        """Approves the Expense and creates accounting entries."""
+        expense.submit_expenses()
+        expense.approve_expenses()
+        expense.action_move_create()
+
+    def test_security(self):
+        # User 2 is only assigned to Operating Unit B2C, and cannot
+        # Access Expenses of Main Operating Unit.
+        record = self.hr_expense_model.sudo(self.user2.id).search(
```

Comment:
It would be good to expand the test with creation or unlink. Following the fix proposed by @sbidoul